### ISSUE TITLE
SupRsync does not update md5sum

### ISSUE THREAD

**Issue by kmharrington** [simonsobs/socs#475]
Thing I've found while testing on the E2E system that we've been avoiding running into because pysmurf-monitor doesn't add files to the database until they're complete. 

After a file is added to the database, suprsync never updates the md5sum values of either the local or the remote files. This means that it never notices it needs to re-copy the file or (worse) if a file is added to the database at time A, copied after it changes at time B, it never realized the file _is_ done copying. 

I'm not prioritizing fixing this at the moment because I only noticed it trying to catch some edge cases in the finalization. But this does mean we are depending on pysmurf-monitor's behavior never changing.

**Comment by jlashner** [comment]
Yeah I agree this would probably be a useful feature to add. The suprsync does update the remote-md5sum each time it attempts to copy a file, but if the remote file changes on its own the suprsync db will not reflect that. I don't think this is a bug since I'm not sure we want to be md5summing our entire archive regularly. But I think some amount of monitoring the state of local-files should probably happen like once an hour or so to protect against cases where the original file was modified or moved after being added to the db.

**Comment by BrianJKoopman** [comment]
> ... I'm not sure we want to be md5summing our entire archive regularly. But I think some amount of monitoring the state of local-files should probably happen like once an hour or so to protect against cases where the original file was modified or moved after being added to the db.

Maybe this is already done, but perhaps file modification times could be checked before computing the md5sum to save some time.
